modules/order/src/Form/UseMultiProfileConfirmForm.php

```diff
@@ -0,0 +1,390 @@
+    $profile_type_storage = $this->entityTypeManager->getStorage('profile_type');
+    /** @var \Drupal\profile\Entity\ProfileTypeInterface $billing_profile_type */
+    $billing_profile_type = $profile_type_storage->load('customer_billing');
+
+    // Import YAML config.
+    $config_path = drupal_get_path('module',
+        'commerce_order') . '/config/shipping_billing_profiles';
+    $source = new FileStorage($config_path);
+    $config_storage = $this->configStorage;
+
+    if (!$billing_profile_type) {
+      $billing_configs = [
+        'core.entity_form_display.profile.customer_billing.default',
+        'core.entity_view_display.profile.customer_billing.default',
+        'field.field.profile.customer_billing.address',
+        'profile.type.customer_billing',
+      ];
+      foreach ($billing_configs as $config_name) {
+        $config_storage->write($config_name, $source->read($config_name));
+      }
+    }
+
+    /** @var \Drupal\profile\Entity\ProfileTypeInterface $shipping_profile_type */
+    $shipping_profile_type = $profile_type_storage->load('customer_shipping');
+
+    if (!$shipping_profile_type) {
+      $shipping_configs = [
+        'core.entity_form_display.profile.customer_shipping.default',
+        'core.entity_view_display.profile.customer_shipping.default',
+        'field.field.profile.customer_shipping.address',
+        'profile.type.customer_shipping',
+      ];
+      foreach ($shipping_configs as $config_name) {
+        $config_storage->write($config_name, $source->read($config_name));
+      }
+    }
+
+    if (!$billing_profile_type || !$shipping_profile_type) {
+      $this->messenger()->addMessage($this->t('Billing and shipping profile
+        types have been successfully created.'
+      ));
+    }
+  }
+
+  /**
+   * Migrate the fields on the customer profile to the newly created types.
+   *
+   * We only migrate the user-defined fields added to the customer profile type
+   * to the billing and shipping profile types.
+   */
+  protected function migrateProfileFields() {
+    // Grab the field definitions from the customer profile type.
+    $field_definitions = $this->entityFieldManager->getFieldDefinitions('profile', 'customer');
+
+    // Let's copy the fields in the 'customer' profile to the billing/shipping
+    // profile types.
+    $profile_bundles = [
+      'customer_billing',
+      'customer_shipping',
+    ];
+    foreach ($profile_bundles as $bundle) {
+      // Grab the field definitions from the customer profile type.
+      $existing_field_definitions = $this->entityFieldManager->getFieldDefinitions('profile', $bundle);
+
+      foreach ($field_definitions as $field_name => $field_definition) {
+        /** @var \Drupal\Core\Field\FieldDefinitionInterface $field_definition */
+        // Don't copy the base fields and the address field.
+        if ($field_definition->getFieldStorageDefinition()->isBaseField() == TRUE) {
+          continue;
+        }
```

Comment:
**OPTIONAL CHANGE REQUEST**

Comparing to TRUE is rather redundant, that's what the `if` statement would do anyway.